deployments/pulse3d/services/pulse3d/src/main.py

```diff
@@ -452,9 +452,12 @@ async def create_new_job(
                         error="AuthorizationError",
                     )
 
-            # second, check usage quota for customer account
+            # second, check usage quota for customer account and if pulse3d version is valid
             usage_quota = await check_customer_quota(con, customer_id, service)
-            if usage_quota["jobs_reached"]:
+            pulse3d_version_status = await con.fetchrow(
+                "SELECT state from pulse3d_versions WHERE version = $1", details.version
+            )
+            if usage_quota["jobs_reached"] or pulse3d_version_status == "removed":
```

Comment:
instead of `pulse3d_version_status == "removed"` we should check `pulse3d_version_status == "deprecated"` and if we've hit the EOL date

Reply:
I think adding this conditional here can add really unexpected behaviors for handling usage restrictions, I think it needs to be it's own check with it's own response.